### PR TITLE
V0.2 vm oversub docs

### PR DESCRIPTION
**Pull request by kyu-kuanwei**
# Description

1. Add oversubscription docs.
2. Add overload docs.

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [x] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: docs/source/scenarios/vm_scheduling.rst [hunk 1]
```diff
@@ -67,15 +67,28 @@ There are two types of meaningful actions:
...
+To maximize each PM's utilization, cloud providers will oversubscribe the physical resource.
+The oversubscription rate can be set in the config.yml. In our scenario, there are two resources
+could be oversubscribed, CPU and memory.
+
+* ``MAX_CPU_OVERSUBSCRIPTION_RATE``: The oversubscription rate of CPU.
+* ``MAX_MEM_OVERSUBSCRIPTION_RATE``: The oversubscription rate of memory.
+
+For example, the default setting of ``MAX_CPU_OVERSUBSCRIPTION_RATE`` is 1.15, that means each
+PM can be allocated at most 1.15 times of its resource capacity.
+
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:82]: Considering the various service level, the physical machines are then divided into the ones over-subscribable and ...

File: docs/source/scenarios/vm_scheduling.rst [hunk 2]
```diff
@@ -67,15 +67,28 @@ There are two types of meaningful actions:
...
+~~~~~~~~~~~~~~~~~~~~
+To maximize each PM's utilization, cloud providers will oversubscribe the physical resource.
+The oversubscription rate can be set in the config.yml. In our scenario, there are two resources
+could be oversubscribed, CPU and memory.
+
+* ``MAX_CPU_OVERSUBSCRIPTION_RATE``: The oversubscription rate of CPU.
+* ``MAX_MEM_OVERSUBSCRIPTION_RATE``: The oversubscription rate of memory.
+
+For example, the default setting of ``MAX_CPU_OVERSUBSCRIPTION_RATE`` is 1.15, that means each
+PM can be allocated at most 1.15 times of its resource capacity.
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:81]: Also add introduction of MAX_...UTIL

File: docs/source/scenarios/vm_scheduling.rst [hunk 3]
```diff
@@ -92,42 +105,62 @@ Energy Curve
...
+As we mention before, the lower energy consumption of the PMs, the lower cost to maintain the physical
 servers. In our simulation, we currently use a non-linear energy curve like the one in the above
 `figure <https://dl.acm.org/doi/10.1145/1273440.1250665>`_ to
-simulate the energy based on the CPU utilization. 
+simulate the energy based on the CPU utilization.
+
+Overload
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Since the VM's CPU utilization varies along the time, when enabling the oversubscription, it might
+happen that the VM's CPU usage execeeds the capacity of the physical resource. This situation called
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:116]: the sum of ?

File: docs/source/scenarios/vm_scheduling.rst [hunk 4]
```diff
@@ -92,42 +105,62 @@ Energy Curve
...
+overload. The threshold of overloading, or we can say the maximum utilization rate can be set in config.yml.
+
+* ``MAX_UTILIZATION_RATE``: The default setting is 1, which means that when the PM's CPU utilization
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:119]: A kind of security mechanism, not used for the judgement

File: docs/source/scenarios/vm_scheduling.rst [hunk 5]
```diff
@@ -92,42 +105,62 @@ Energy Curve
...
+  exceeds 100%, it would be considered as overloading.
+
+Overloading may lead to VM's performance degradation or service level agreements (SLAs) violations
+in real production (We will support these features in the future).
+Currently, we only support quiescing (killing) all VMs, which can also be set in config.yml.
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:124]: Currently, for the situation of overloading, we only support quiescing (killing) all VMs or just recording the times of overloading, which can also be set in config.yml.

File: docs/source/scenarios/vm_scheduling.rst [hunk 6]
```diff
@@ -92,42 +105,62 @@ Energy Curve
...
 
+* ``KILL_ALL_VMS_IF_OVERLOAD``: If this action is enable,
+  once overloading happens, all VMs located at the overloading PMs will be deallocated. To consider the
+  effect of overloading, we will calculate the updated utilization (=0)
```
Comment (Jinyu-W) [inline docs/source/scenarios/vm_scheduling.rst:128]: To consider the effect of overloading, we will still count the energy consumed by the high utilization. The impact of the quiescing action on the PM's utilization will be reflected in the next tick.

File: docs/source/scenarios/vm_scheduling.rst (hunk 1 above)
Comment (kyu-kuanwei) [reply docs/source/scenarios/vm_scheduling.rst:82]: Shall we also introduce about the VM category (PM will be non-oversubscribable if one of its vm is interactive)?

Comment (Jinyu-W) [reply docs/source/scenarios/vm_scheduling.rst:82]: Good point, introduce it in `VM Request` part?

**Review by Jinyu-W — APPROVED** [review]

**Pull request merged**